fhir-model/src/main/java/com/ibm/fhir/model/util/ValidationSupport.java

```diff
@@ -285,6 +285,10 @@ public static void checkValue(String value, Pattern pattern) {
                 List<String> typeNameList = Arrays.stream(types).map(Class::getSimpleName).collect(Collectors.toList());
                 throw new IllegalStateException(String.format("Invalid type: %s for choice element: '%s' must be one of: %s", elementType.getSimpleName(), elementName, typeNameList.toString()));
             }
+            //do other validation here?
```

Comment:
I don't think you need to do this here since you're generating the ValidationSupport.checkReferenceType() call for reference-type choice elements now

Reply:
good call, meant to remove this.
I actually think there are other validation cases (for both choice types and list types) that we're still missing, but in this PR I was focused on just the referenceType-checking

Reply:
removed